src/js/client/templateFormControl.js

```diff
@@ -0,0 +1,180 @@
+/*
+
+  A component to standardize handling of "simple" form data submitted via AJAX.  This component:
+
+  1.  Performs an initial render of the component using the template specified in `options.templates.initial`
+  2.  Binds a form submission that sends the `model` data formatted using `options.rules.request` using the options
+      specified in `options.ajax`
+
+  For more details on the request and response cycle, see the `templateRequestAndUpdate` grade.
+
+ */
+/* global fluid, jQuery */
+(function () {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.templates.templateFormControl");
+
+
+    // TODO: Replace this with JSON Schema validation: https://issues.gpii.net/browse/GPII-1176
+    gpii.templates.templateFormControl.checkRequirements = function (that) {
+        var errors = [];
+
+        if (!that.options.templates || !that.options.templates.initial || !that.options.templates.error || !that.options.templates.success) {
```

Comment:
There is now a common grade to enable required option checking on startup.  All previous copypasta has been eliminated in favor of this new grade.